espnet2/asr_transducer/decoder/rnn_decoder.py

```diff
@@ -0,0 +1,262 @@
+"""RNN decoder definition for Transducer models."""
+
+from typing import List, Optional, Tuple, Union
+
+import torch
+from typeguard import check_argument_types
+
+from espnet2.asr_transducer.beam_search_transducer import ExtendedHypothesis, Hypothesis
+from espnet2.asr_transducer.decoder.abs_decoder import AbsDecoder
+
+
+class RNNDecoder(AbsDecoder):
+    """RNN decoder module.
+
+    Args:
+        vocab_size: Vocabulary size.
+        embed_size: Embedding size.
+        hidden_size: Hidden size..
+        rnn_type: Decoder layers type.
+        num_layers: Number of decoder layers.
+        dropout_rate: Dropout rate for decoder layers.
+        embed_dropout_rate: Dropout rate for embedding layer.
+        embed_pad: Embedding padding symbol ID.
+
+    """
+
+    def __init__(
+        self,
+        vocab_size: int,
+        embed_size: int = 256,
+        hidden_size: int = 256,
+        rnn_type: str = "lstm",
+        num_layers: int = 1,
+        dropout_rate: float = 0.0,
+        embed_dropout_rate: float = 0.0,
+        embed_pad: int = 0,
+    ) -> None:
+        assert check_argument_types()
+
+        if rnn_type not in {"lstm", "gru"}:
```

Comment:
```suggestion
        if rnn_type not in ("lstm", "gru"):
```
A tuple may be more efficient than a set here.